lib/internal/event_target.js

```diff
@@ -0,0 +1,392 @@
+'use strict';
+
+const {
+  ArrayFrom,
+  Object,
+  Map,
+  SafeWeakMap,
+  Set,
+  SymbolFor,
+} = primordials;
+
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE,
+    ERR_INVALID_THIS,
+    ERR_EVENT_RECURSION,
+  }
+} = require('internal/errors');
+const kRejection = SymbolFor('nodejs.rejection');
+
+const perf_hooks = require('perf_hooks');
+const { customInspectSymbol } = require('internal/util');
+const { inspect } = require('util');
+
+const events = new SafeWeakMap();
+const targets = new SafeWeakMap();
+
+class Event {
+  constructor(type, options) {
+    if (typeof type !== 'string')
+      throw new ERR_INVALID_ARG_TYPE('type', 'string', type);
```

Comment:
Are we allowed to throw these errors if we ever want to be spec compliant with the DOM?

Reply:
We could easily copy the browser's behavior here and coerce the type to a string.

Reply:
I think that would make writing universal code easier and I'm usually not the one to push-back on stuff that is not 100% spec compliant.

Reply:
Done :-)